From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LessOS Build System <build@lessos>
Date: Sun, 29 Dec 2024 18:45:00 +0000
Subject: [PATCH] Fix linking with duplicate parted libraries

Both libparted and libparted-fs-resize have overlapping symbols.
Use --allow-multiple-definition to permit this, and ensure we have
all required libraries (uuid, blkid).
---
 Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 1111111..2222222 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,7 +5,8 @@ man_MANS = fatresize.1
 fatresize_CPPFLAGS = $(PARTED_CFLAGS)

 fatresize_SOURCES = fatresize.c
-fatresize_LDADD = $(PARTED_LIBS) -luuid -lblkid
+fatresize_LDADD = $(PARTED_LIBS) -luuid -lblkid
+fatresize_LDFLAGS = -Wl,--allow-multiple-definition

 EXTRA_DIST = AUTHORS CREDITS ChangeLog NEWS README \
 	fatresize.spec $(man_MANS)
--
2.43.0
